src/lib/repositories/impl/product_repository_impl.py

```diff
@@ -10,19 +12,39 @@ def __init__(self):
 
     def add(self, product):
         product.id = self._current_id
+        product.create_date = datetime.now()
         self._products[product.id] = product
         self._current_id += 1
 
     def get_by_id(self, product_id):
-        return self._products[product_id]
+        product_to_return = self._products[product_id]
+        product_filtered = list(
+            filter(
+                lambda product: product.entity_status == AuditStatus.ACTIVE,
+                [product_to_return],
+            )
+        )
+        return product_filtered[0]
 
     def get_all(self):
-        return list(self._products.values())
-
-    def delete_by_id(self, product_id):
-        self._products.pop(product_id)
+        products = list(self._products.values())
+        products_filtered = list(
+            filter(
+                lambda product: product.entity_status == AuditStatus.ACTIVE, products
+            )
+        )
+        return list(products_filtered)
+
+    def delete_by_id(self, product_id, product):
+        product.delete_date = datetime.now()
```

Comment:
did you mean `updated_date`?

Also can you do a `get_by_id` here? the only thing you need right now from `product` is the `updated_by`.

Should look like:

```
product_to_delete = self.get_by_id(product_id)
product_to_delete.entity_status = DELETED
product_to_delete.updated_date = now()
product_to_delete.updated_by = product.updated_by
self.update_by_id(product_id, product_to_delete)
```

Reply:
done.